app/controllers/search_controller.rb

```diff
@@ -5,6 +5,9 @@ class SearchController < ApplicationController
   respond_to :json
 
   def extract_filters
+    isImmediateReentered = (params.delete :rerun) == 'true'
```

Comment:
this variable should be named `is_immediate_reentered`, along with `previous_keyword` below